verilog/tools/lint/BUILD

```diff
@@ -109,15 +110,15 @@ _linter_test_configs = [
             ],
             expect_fail = True,
         ),
-        # Verifies that all rules *except* this one is lint clean.
-        verilog_style_lint.test(
-            name = testfilebase + "-minusrule-test",
-            srcs = ["testdata/{}.sv".format(testfilebase)],
-            flags = [
-                "--ruleset=all",
-                "--rules=-" + rule,  # exclude one rule
-            ],
-        ),
+#        # Verifies that all rules *except* this one is lint clean.
```

Comment:
mmh, was this needed due to the overlap of the two always-ff rules ?
Maybe we need to have the files more specific.

Reply:
This mechanism ensures that the responsibilities of linter rules are mutually exclusive. As detailed in the above description, this would no longer be the case if both the original `always_ff_non_blocking_rule` and this `always_ff_only_local_blocking_rule` were to be offered as alternatives. Their responsibilities overlap.
From the preceding discussion, I assume that a full replacement of the original rule is not desired. So, I can identify to remaining options to proceed:
1. Relax the exclusiveness requirements, or
2. Merge the rules into one and make its aggressiveness selectable by some parameter (I am not aware of a mechanism to do so.)

Reply:
I am for merging the rules and make them configurable so that we end up with only one rule. As a first step having something that can be configured so that it behaves as before, but later we might consider to always keep the more stricter form (this is to help transitioning people using the weaker lint rules so far).

There is a simple way to provide configuration to rules. Each rule can override a `Configure(absl::string_view configuration)` method which receives a string that contains configurations.

There are already utilities that help and set configuration parameters, see `common/text/config_utils.h` for this. These allow to directly set a value inside the rule with a very simple set-up.

Here are some examples of rule configurations
  * [LineLengthRule](https://github.com/google/verible/blob/00b99903ae110367e55f8f249d1366cd6c9cf4d6/verilog/analysis/checkers/line_length_rule.cc#L186-L191)
  * [ModuleFilenameRule](https://github.com/google/verible/blob/00b99903ae110367e55f8f249d1366cd6c9cf4d6/verilog/analysis/checkers/module_filename_rule.cc#L133-L137)
  * [ExplicitParameterStorageRule](https://github.com/google/verible/blob/00b99903ae110367e55f8f249d1366cd6c9cf4d6/verilog/analysis/checkers/explicit_parameter_storage_type_rule.cc#L106-L116)
  * [ForbiddenAnonymousStructsUnionsRule](https://github.com/google/verible/blob/00b99903ae110367e55f8f249d1366cd6c9cf4d6/verilog/analysis/checkers/forbidden_anonymous_structs_unions_rule.cc#L69-L75)
  * [ExplicitParameterStorageRule](https://github.com/google/verible/blob/00b99903ae110367e55f8f249d1366cd6c9cf4d6/verilog/analysis/checkers/explicit_parameter_storage_type_rule.cc#L106-L116)

For the `verilog_style_lint.test()`, they would just run in the default configuration, so the existing test should trigger there.

Inside the unit-tests, you can test the behaviour using the [`RunConfiguredLintTestCases<>`](https://github.com/google/verible/blob/00b99903ae110367e55f8f249d1366cd6c9cf4d6/common/analysis/linter_test_utils.h#L91) call. It is easy to see in the existing examples, such as in the [line length rule test](https://github.com/google/verible/blob/00b99903ae110367e55f8f249d1366cd6c9cf4d6/verilog/analysis/checkers/line_length_rule_test.cc#L198-L199)